future/src/ct/ct_storage_control.cc

```diff
@@ -28,7 +28,17 @@
 #include <glib/gstdio.h>
 
 
-static CtStorageEntity* get_entity_by_type(CtMainWin* pCtMainWin, CtDocType file_type)
+Glib::ustring CtStorageControl::_check_and_unpack(const Glib::ustring& path) {
+    // unpack file if need
+    Glib::ustring password;
```

Comment:
also the password is just a string of plain bytes to be passed to p7zip so std::string is the right container